src/lib/parser/cli.ts

```diff
@@ -231,21 +239,143 @@ async function watchMode (input: string | MgbaWebSocketClient, options: { debug:
       // Only update display if party data changed or first run
       if (dataHash !== lastDataHash || isFirstRun) {
         clearScreen()
-        // Only show party summary - no other logs
-        displayPartyPokemon(result.party_pokemon, typeof input === 'string' ? 'FILE' : 'MEMORY')
+        displayPartyPokemon(result.party_pokemon, 'FILE')
 
         lastDataHash = dataHash
         isFirstRun = false
       }
     } catch (error) {
-      // Silent error handling - don't clear screen or show errors unless critical
       console.error('❌ Error:', error instanceof Error ? error.message : 'Unknown error')
     }
 
     await new Promise(resolve => setTimeout(resolve, options.interval))
   }
 }
 
+/**
+ * WebSocket-based watch mode - event-driven approach using memory change listeners
+ */
+async function watchModeWebSocket (client: MgbaWebSocketClient, options: { debug: boolean, graph: boolean, interval: number }) {
+  console.log('🔄 Starting event-driven watch mode...')
+  console.log('Setting up memory region watching for real-time updates')
+  console.log('Press Ctrl+C to exit')
+
+  // Create parser once and reuse it
+  const parser = new PokemonSaveParser()
+  let lastDataHash = ''
+  let isFirstRun = true
+
+  // Set up memory change listener for real-time updates
+  const memoryChangeListener = async (address: number, _size: number, _data: Uint8Array) => {
+    try {
+      // Only process changes to party-related memory regions
+      if (address === 0x20244e9 || address === 0x20244ec) {
+        // Parse the updated save data
+        const result = await parser.parseSaveFile(client)
+
+        // Create a simple hash of the party data to detect changes
+        const dataHash = JSON.stringify(
+          result.party_pokemon.map(p => ({
+            species: p.speciesId,
+            level: p.level,
+            hp: p.currentHp,
+            nickname: p.nickname,
+          })),
+        )
+
+        // Only update display if party data actually changed
+        if (dataHash !== lastDataHash || isFirstRun) {
+          clearScreen()
+          console.log(`🔄 Party data updated at 0x${address.toString(16)} (${new Date().toLocaleTimeString()})`)
```

Comment:
the watching should be integrated in PokemonSaveParser instead. we do not want to do this manually in our codes. the PokemonSaveParser should offer parser apis that let us decide if it should be watching something and offer callbacks or events. maybe parser.watch(...)